src/encryption/encryption.ts

```diff
@@ -41,63 +42,61 @@ export interface IEncryptionResult {
+ * @deprecated This method should be replaced by encryptWithGeneratedKey.
+ * This method convert give data string to UTF-8 bytes UTF-8 bytes
  */
-export async function encryptStringWithGeneratedKey(
-  options: IEncryptionOptionsWithoutKey,
-  serializationVersion: SerializationFormat = SerializationFormat.latest_version
-): Promise<IEncryptionResult & { generatedKey: string }> {
-  return encryptWithGeneratedKey(options, serializationVersion, 'utf8');
+export async function encryptStringWithGeneratedKey({
+  data,
+  strategy,
+  iv,
+  serializationVersion = SerializationFormat.latest_version,
+}: {
+  data: string;
+  strategy: CipherStrategy;
+  iv?: string;
+  serializationVersion?: SerializationFormat;
+}): Promise<IEncryptionResult & { generatedKey: string }> {
+  return encryptWithGeneratedKey({
+    options: { data: utf8ToBytes(data), strategy, iv },
+    serializationVersion,
+  });
 }
 
 /**
- * Similar to `encryptBinaryWithKey`
- * but generates random bytes to use as the key. This will be returned with the result.
+ * @deprecated This method should be replaced by encryptWithGeneratedKey.
+ * This method convert give data string to UTF-8 bytes UTF-8 bytes
  */
-export async function encryptBinaryWithGeneratedKey(
-  options: IEncryptionOptionsWithoutKey,
-  serializationVersion: SerializationFormat = SerializationFormat.latest_version
-): Promise<IEncryptionResult & { generatedKey: string }> {
-  return encryptWithGeneratedKey(options, serializationVersion, 'raw');
+export async function encryptBinaryWithGeneratedKey({
+  data,
+  strategy,
+  iv,
+  serializationVersion = SerializationFormat.latest_version,
+}: {
+  data: string;
+  strategy: CipherStrategy;
+  iv?: string;
+  serializationVersion?: SerializationFormat;
+}): Promise<IEncryptionResult & { generatedKey: string }> {
+  return encryptWithGeneratedKey({
+    options: { data: binaryStringToBytes(data), strategy, iv },
+    serializationVersion,
+  });
 }
 
-/**
- * @deprecated This method should be replaced by
- * encryptBinaryWithGeneratedKey or encryptStringWithGeneratedKey method.
- * The default encoding is 'raw' is suitable for binaries and 1 byte UTF-8.
- * string with greater than 2 bytes UTF-8 string will produce an incorrect result. e.g. data string '鍵键'
- * encrypted with 'raw' encoding will produce incorrect decrypted value.
- */
-export async function encryptWithGeneratedKey(
-  options: IEncryptionOptionsWithoutKey,
-  serializationVersion: SerializationFormat = SerializationFormat.latest_version,
-  encoding: Encoding = 'raw'
-): Promise<IEncryptionResult & { generatedKey: string }> {
-  const key = generateRandomKey(options.keyLength || 32);
+export async function encryptWithGeneratedKey({
+  options,
+  serializationVersion = SerializationFormat.latest_version,
+}: {
+  options: IEncryptionOptionsWithoutKey;
+  serializationVersion?: SerializationFormat;
+}): Promise<IEncryptionResult & { generatedKey: string }> {
```

Comment:
generatedKey should probably be an `EncryptionKey`

Reply:
done